src/javascript/app/pages/trade/charts/digit_info.js

```diff
@@ -249,11 +249,13 @@ const DigitInfo = (() => {
         if (stream_id) {
             if (chart.series[0].name === tick.tick.symbol) {
                 stream_id = tick.tick.id || null;
-                update(tick.tick.symbol, tick.tick.quote);
+                update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
             } else {
                 BinarySocket.send({ forget: (tick.tick.id).toString() });
+                stream_id = null;
             }
         } else {
+            stream_id = tick.tick.id || null;
```

Comment:
`stream_is = tick.tick.id || null `and `update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',','')` are used in both conditions. Can we rework this if-else structure?

Reply:
there are three conditions

Reply:
Something like this, but 100% I missed something))))
```
    const updateChart = async (tick) => {
        const decimal_places = await Symbols.getUnderlyingPipSize(tick.tick.symbol);
        stream_id = tick.tick.id || null;
        if (stream_id && chart.series[0].name !== tick.tick.symbol) {
                BinarySocket.send({ forget: (tick.tick.id).toString() });
                stream_id = null;
        } else {
            update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
        }
    };
```

Reply:
Please, ignore if irrelevant

Reply:
your else doesnt trigger when stream_id is null

Reply:
how about this one? :)
```
            if (chart.series[0].name === tick.tick.symbol) {
                stream_id = tick.tick.id || null;
                update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
            } else if (stream_id) {
               BinarySocket.send({ forget: (tick.tick.id).toString() });
                stream_id = null;
            }
```

Reply:
i think this one ignores the case when chart.series[0].name !== tick.tick.symbol

quite delicate the logic on this.

Essentially i'll likely just end up with if elsif and else.

Reply:
ok, here it is :)):
```
            if (!stream_id || chart.series[0].name === tick.tick.symbol) {
                stream_id = tick.tick.id || null;
                update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
            } else if (stream_id) {
               BinarySocket.send({ forget: (tick.tick.id).toString() });
                stream_id = null;
            }
```

Reply:
i think else if doesnt trigger when only chart.series[0].name !== tick.tick.symbol is true right

Reply:
im more confused now than before lol

Reply:
in the original condition this part:
```
BinarySocket.send({ forget: (tick.tick.id).toString() });
stream_id = null;
```
should be triggered when `stream_id` is `true` and `chart.series[0].name === tick.tick.symbol` is false at the same time

Reply:
@maryia-deriv let's not touch what has been working for many years. There's more than one way how to do stuff, programming is not an exact science)